extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -111,10 +112,26 @@ export class AmpViewerIntegration {
         }
       );
     }
-    /** @type {?HighlightInfoDef} */
-    const highlightInfo = getHighlightParam(ampdoc);
-    if (highlightInfo) {
-      this.highlightHandler_ = new HighlightHandler(ampdoc, highlightInfo);
+
+    if (
+      'fragmentDirective' in document &&
+      isExperimentOn(this.win, 'enable-text-fragments')
+    ) {
+      this.win.addEventListener('message', (e) => {
+        if (
+          e.origin !== 'https://www.google.com' ||
+          e.data?.directive?.length === 0
+        ) {
+          return;
+        }
+        this.updateUrlWithTextFragment_(e.data.directive);
```

Comment:
Nit: I'd just inline this instead of making a helper.

Reply:
This is a workaround sinon.js not allowing to stub out `window.location.replace` (errors with `Cannot redefine property: replace`)